src/Shared/CertificateGeneration/CertificateFolderStore.cs

```diff
@@ -0,0 +1,107 @@
+
+        public CertificateFolderStore(string name, string path, ProcessRunOptions updateStore) :
+            base(name)
+        {
+            FolderPath = path;
+            _updateStore = updateStore;
+        }
+
+        private void ReportElevationNeeded(IReporter? reporter)
+        {
+            if (Elevate)
+            {
+                reporter?.Output($"Changing '{StoreName}' requires root priviledges. You may be prompted for your password.");
+            }
+        }
+
+        public override bool CheckDependencies(IReporter? reporter)
+        {
+            return CheckProgramDependency(_updateStore, reporter);
+        }
+
+        public override bool TryInstallCertificate(string name, PemCertificateFile pemFile, IReporter? reporter, bool isInteractive)
+        {
+            ReportElevationNeeded(reporter);
+            CopyFile(pemFile.FilePath, GetCertificatePath(name), isInteractive);
+            ProcessRunner.Run(_updateStore with { IsInteractive = isInteractive });
+            return true;
+        }
+
+        public override void DeleteCertificate(string name, IReporter? reporter, bool isInteractive)
+        {
+            ReportElevationNeeded(reporter);
+            DeleteFile(GetCertificatePath(name), isInteractive);
+        }
+
+        public override bool HasCertificate(string name, X509Certificate2 certificate)
+        {
+            string certificatePath = GetCertificatePath(name);
+            if (!File.Exists(certificatePath))
+            {
+                return false;
+            }
+            X509Certificate2 storeCertificate = X509Certificate2.CreateFromPem(File.ReadAllText(certificatePath));
+            return storeCertificate.Equals(certificate);
+        }
+
+        private string GetCertificatePath(string name)
+            => Path.Combine(FolderPath, name + ".pem");
+
+        private void DeleteFile(string path, bool isInteractive)
+        {
+            if (!File.Exists(path))
+            {
+                return;
+            }
+
+            if (Elevate)
+            {
+                ProcessRunner.Run(new()
+                {
+                    Command = { "rm", path },
+                    Elevate = true,
+                    IsInteractive = isInteractive
+                });
+            }
+            else
+            {
+                File.Delete(path);
+            }
+        }
+
+        private void CopyFile(string sourceFileName, string destFileName, bool isInteractive)
+        {
+            DeleteFile(destFileName, isInteractive);
+
+            if (Elevate)
+            {
+                ProcessRunner.Run(new()
+                {
+                    Command = { "cp", sourceFileName, destFileName },
+                    Elevate = true,
+                    IsInteractive = isInteractive
+                });
+            }
+            else
+            {
+                File.Copy(sourceFileName, destFileName);
+            }
+        }
+    }
```

Comment:
We should just use C# for these operations. 

By the time we get here we should already make sure we are running with admin rights if we need to.

Reply:
> By the time we get here we should already make sure we are running with admin rights if we need to.

This is similar to the mac implementation which uses `sudo` for performing specific operations that need root.
Do you want to do something different for Linux?